resources/migrations/001_clojure_migration.clj

```diff
@@ -0,0 +1,31 @@
+(ns migrations.001-clojure-migration)
+
+(defmacro defmigration
+  ([name execute-thunk]
+   `(defmigration ~name ~execute-thunk nil))
+  ([name execute-thunk rollback-thunk]
+   `(do
+      (defn ~(symbol (str name "-execute")) [~'database] (~execute-thunk ~'database))
+      ~(when (seq rollback-thunk)
+         `(defn ~(symbol (str name "-rollback")) [~'database] (~rollback-thunk ~'database))))))
```

Comment:
I don't think we need or want a macro for this. Just give it the name a function to run for the up migration and one for the down migration.